rust_src/src/frames.rs

```diff
@@ -56,4 +62,71 @@ pub fn selected_frame() -> LispObject {
     unsafe { LispObject::from(current_frame) }
 }
 
+/// Return non-nil if OBJECT is a frame which has not been deleted.
+/// Value is nil if OBJECT is not a live frame.  If object is a live
+/// frame, the return value indicates what sort of terminal device it is
+/// displayed on.  See the documentation of `framep' for possible
+/// return values.
+#[lisp_fn]
+fn frame_live_p(object: LispObject) -> LispObject {
+    if object.as_frame().map_or(false, |f| f.is_live()) {
+        LispObject::from(framep(object))
+    } else {
+        LispObject::constant_nil()
+    }
+}
+
+/// Set selected window of FRAME to WINDOW.
+/// FRAME must be a live frame and defaults to the selected one.  If FRAME
+/// is the selected frame, this makes WINDOW the selected window.  Optional
+/// argument NORECORD non-nil means to neither change the order of recently
+/// selected windows nor the buffer list.  WINDOW must denote a live window.
+/// Return WINDOW.
+#[lisp_fn(min = "2")]
+pub fn set_frame_selected_window(
+    frame: LispObject,
+    window: LispObject,
+    norecord: LispObject,
+) -> LispObject {
+    let f = if frame.is_nil() {
+        selected_frame()
+    } else {
+        frame
+    };
+    let mut frame_ref = f.as_live_frame_or_error();
+    let w = window.as_live_window_or_error();
+
```

Comment:
I have to do it this way as I need the updated frame `LispObject` below.

Reply:
`frame_live_or_selected` landed while this was in the works. That simplifies the logic here.